### PR TITLE
Make meter receiver support MAL

### PR DESCRIPTION
**Pull request by mrproliu**
### Make agent side meter receiver support MAL
- [x] Update the documentation to include this new feature.
- [x] Tests(including UT, IT, E2E) are added to verify the new feature.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

**Comment by wu-sheng** [comment]
FYI @ascrutae @hailin0 as you are doing something relating to this.

File: oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/dsl/SampleFamily.java [hunk 1]
```diff
@@ -233,12 +234,29 @@ public SampleFamily tag(Closure<?> cl) {
         }).toArray(Sample[]::new));
     }
 
+    /**
+     * Ignore histogram decrease bucket value, at agent side meter, need to ignore decrease histogram bucket value
+     */
+    public SampleFamily ignoreHistogramDecrease() {
```
Comment (hanahmily) [inline oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/dsl/SampleFamily.java:240]: From https://en.wikipedia.org/wiki/Histogram#Mathematical_definition, there are two types of histogram bin:
 * Cumulative: prometheus, otel.
 * Ordinary: agent meter.

Let's provide two functions to set it, and defaults to cumulative.

File: oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/dsl/SampleFamily.java [hunk 2]
```diff
@@ -233,12 +234,29 @@ public SampleFamily tag(Closure<?> cl) {
...
+        this.context.ignoreHistogramDecrease = true;
+        return this;
+    }
+
+    /**
+     * Change default histogram bucket value unit,
+     * use second on the Prometheus, use millisecond on the meter side meter
+     */
+    public SampleFamily defaultHistogramBucketUnit(TimeUnit unit) {
```
Comment (hanahmily) [inline oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/dsl/SampleFamily.java:249]: My preference is to add a new factory to build SampleFamily with those two settings. If we define functions in SampleFamily, that means end-users could access them, that will cause some potential issues. 

and they represent the different behaviors of the eco-system, I don't find any reason to access them in a MAL expression.

File: oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/dsl/SampleFamily.java (hunk 1 above)
Comment (hanahmily) [reply oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/dsl/SampleFamily.java:240]: What I mean is `histogramDecrease` is not clear and very confusing to ppl. Let's opt for the notions that is well-known for ppl.

Comment (mrproliu) [reply oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/dsl/SampleFamily.java]: Created `HistogramType` enum to adapt this.

File: oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/dsl/SampleFamily.java (hunk 2 above)
Comment (mrproliu) [reply oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/dsl/SampleFamily.java]: Have created `SampleFactoryBuilder` to help create `SampleFactory` with these two settings.

File: oap-server/analyzer/agent-analyzer/pom.xml [hunk 3]
```diff
@@ -39,8 +39,9 @@
             <version>${project.version}</version>
         </dependency>
         <dependency>
-            <groupId>org.codehaus.groovy</groupId>
-            <artifactId>groovy</artifactId>
+            <groupId>org.apache.skywalking</groupId>
```
Comment (kezhenxu94) [inline oap-server/analyzer/agent-analyzer/pom.xml:42]: this is the only place where groovy is used? if so, you need to update/remove the license and the dependency in the `oap-server/pom.xml`

**Comment by wu-sheng** [comment]
@mrproliu Could you consider to write a blog about the MAL?

**Comment by mrproliu** [comment]
> @mrproliu Could you consider to write a blog about the MAL?

Sure, I may start at the weekend, at that time, I will synchronize the progress with you.

**Review by hanahmily — APPROVED** [review]
LGTM, nice work @mrproliu

**Pull request merged**